CRM/Civicase/Hook/PageRun/ViewCasePageRedirect.php

```diff
@@ -45,9 +45,10 @@ private function redirectViewCasePage($caseId) {
     // Add selected parameters passed to this page to the redirect URL.
     $relevantUrlParams = [['name' => 'tab', 'type' => 'String']];
     $caseCategoryName = CaseCategoryHelper::getCaseCategoryNameFromOptionValue($case['case_type_id.case_type_category']);
-    $fragment = "/case/list?sf=id&sd=DESC&caseId={$caseId}&cf=%7B%22case_type_category%22:%22" . strtolower($caseCategoryName) . "%22,%22status_id%22:%5B%22{$case['status_id']}%22%5D,%22case_type_id%22:%5B%22{$case['case_type_id.name']}%22%5D%7D";
+    $caseCategoryId = CaseCategoryHelper::getCategoryByName($caseCategoryName, 'id');
```

Comment:
This is actually the case category Id: `$case['case_type_id.case_type_category']` so we don't need this function, and the function to fetch the case category name also.